src/SFML/Graphics/GLCheck.hpp

```diff
@@ -53,20 +58,28 @@ bool glCheckError(const std::filesystem::path& file, unsigned int line, std::str
 #ifdef SFML_DEBUG
 // In debug mode, perform a test on every OpenGL call
 // The lamdba allows us to call glCheck as an expression and acts as a single statement perfect for if/else statements
-#define glCheck(...)                                                                             \
-    [](auto&& glCheckInternalFunction)                                                           \
-    {                                                                                            \
-        if constexpr (!std::is_void_v<decltype(glCheckInternalFunction())>)                      \
-        {                                                                                        \
-            const auto glCheckInternalReturnValue = glCheckInternalFunction();                   \
-            sf::priv::glCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-            return glCheckInternalReturnValue;                                                   \
-        }                                                                                        \
-        else                                                                                     \
-        {                                                                                        \
-            glCheckInternalFunction();                                                           \
-            sf::priv::glCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-        }                                                                                        \
+#define glCheck(...)                                                                                     \
+    [](auto&& glCheckInternalFunction)                                                                   \
+    {                                                                                                    \
+        if (const GLenum error = glGetError(); error != GL_NO_ERROR)                                     \
```

Comment:
choose a longer name like `glCheckInternalError` to not cause bugs with shadowing and consistency with the other names in the lamdba